lib/schema.js

```diff
@@ -0,0 +1,92 @@
+const _ = require('lodash')
+const { joiStringRegexConvertible } = require('./joi-extension')
+const Joi = require('joi').extend([joiStringRegexConvertible])
+
+const category = Joi.object().keys({
+  title: Joi.string().required(),
+  label: Joi.string()
+})
+
+const searchReplace = Joi.object().keys({
+  search: Joi.stringRegexConvertible(),
+  replace: Joi.string()
+    .allow('')
+    .required()
+})
+
+const fields = {
+  branches: Joi.array()
+    .items(Joi.string())
+    .single()
+    .description(
```

Comment:
Are these descriptions used in the JOI error output anywhere? I'm a bit worried about the duplication between here and the readme, and if they're actually output then perhaps we can just leave them off?

Reply:
Used in the schema output.